### DISCUSSION TITLE
Connect to database with pgAdmin

### DISCUSSION THREAD

**Discussion by deanfourie1** [netbox-community/netbox#16251] [General]
Hi, I am using pgAdmin running in a docker container.

I am trying to use pgAdmin to better manage the PostgreSQL database as im not great with SQL cli,

I am running netbox on the host machine, not in docker,
I am running pgAdmin in docker,

The problem I am having is that I cannot connect to the HOST database from within the docker container, 

Does anyone know how I can do this? Ive tried with the IP address 172.16.101.2 and localhost / 127.0.0.1 but they all fail to connect.

Here is my docker run,

```
docker pull dpage/pgadmin4
docker run -p 8080:80 \
--add-host host.docker.internal:host-gateway \
    -e 'PGADMIN_DEFAULT_EMAIL=dean@deanfourie.me' \
    -e 'PGADMIN_DEFAULT_PASSWORD=password' \
    -e 'PGADMIN_CONFIG_ENHANCED_COOKIE_PROTECTION=True' \
    -e 'PGADMIN_CONFIG_LOGIN_BANNER="Authorised users only!"' \
    -e 'PGADMIN_CONFIG_CONSOLE_LOG_LEVEL=10' \
    -d dpage/pgadmin4
```

**Comment by marsteel** [comment]
try to use tcpdump on host to capture the PostgreSQL traffic from the pgadmin4 container.

**Comment by deanfourie1** [comment]
I honestly feel like this is a network issue relating to my container running pgAdmin,

It doesnt even really appear to try and fail, just fails. I still dont think the container can reach the host.

**Comment by deanfourie1** [comment]
tcpdump confirmed the capture no packets.

In my docker command for pgAdmin I have added,

`--add-host=host.docker.internal:host-gateway`

<img width="526" alt="image" src="https://github.com/netbox-community/netbox/assets/71812804/eb69e822-b0b4-4781-ba0a-069fc3554437">

**Reply by marsteel** [reply]
There is a dns resolve issue. and the port should be 5433 because your postgresql is listening at 5433

**Comment by deanfourie1** [comment]
Also, looking at PostgreSQL configuration, its using port 5433 for netbox? Can you please confirm this?

Im sure thats not helping!

<img width="571" alt="image" src="https://github.com/netbox-community/netbox/assets/71812804/d4c131c9-dc46-4786-aca1-3cc6d8a1762b">

**Reply by marsteel** [reply]
your postgresql listenning port is 5433

mine is the default 5432
```

# netstat -an | grep 5432
tcp        0      0 127.0.0.1:5432          0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:37366         127.0.0.1:5432          ESTABLISHED
tcp        0      0 127.0.0.1:37360         127.0.0.1:5432          ESTABLISHED
tcp        0      0 127.0.0.1:5432          127.0.0.1:37360         ESTABLISHED
tcp        0      0 127.0.0.1:5432          127.0.0.1:37358         ESTABLISHED
tcp        0      0 127.0.0.1:5432          127.0.0.1:39446         ESTABLISHED
tcp        0      0 127.0.0.1:5432          127.0.0.1:37382         ESTABLISHED
tcp        0      0 127.0.0.1:5432          127.0.0.1:37366         ESTABLISHED
tcp        0      0 127.0.0.1:37358         127.0.0.1:5432          ESTABLISHED
tcp        0      0 127.0.0.1:37382         127.0.0.1:5432          ESTABLISHED
tcp        0      0 127.0.0.1:39446         127.0.0.1:5432          ESTABLISHED
unix  2      [ ACC ]     STREAM     LISTENING     6324809  /var/run/postgresql/.s.PGSQL.5432
```

**Reply by deanfourie1** [reply]
looks like version 13 is on port 5432 and version 15 is on 5433

**Comment by deanfourie1** [comment]
Its weird but I also have a listening socket on 5432. I have active connections to this socket too.

**Comment by deanfourie1** [comment]
I installed pgAdmin the using the non docker method, and it connected to the database on port 5432 first time,

Was definitely the container unable to communication with the host, and I could not resolve it :(.

Hate that.
